### PR TITLE
Finishes [#187355115] adding-products-to-wishlist

### PR DESCRIPTION
**Pull request by igorihimbazwe**
**1. What does this PR do?**

Adding product to wishlist

**2. Description of task to be completed**

_Add Product to Wishlist:_

**Description:** Allows a user to add a product to their wishlist. The user must provide the product ID and be authenticated with a JWT token.
_Remove Product from Wishlist:_

**Description:** Allows a user to remove a product from their wishlist. The user must provide the product ID and be authenticated with a JWT token.
_Get User Wishlist:_

**Description:** Retrieves the list of products in a user's wishlist. The user must be authenticated with a JWT token.

**3. Relevant pivotal tracker stories**

https://www.pivotaltracker.com/story/show/187355115
**4. How should this be manually tested?**
clone the repository and checkout the branch <ft-wishlist-functionality>
Run your server: Ensure your Express server is running.
Access your Swagger UI: Typically, Swagger UI is accessible at http://localhost:8000/docs 

**5. Screenshoots**
Screenshot 2024-05-21 at 3 31 44 PM
Screenshot 2024-05-21 at 3 32 19 PM

### CODE REVIEW

**Review by Celestin4 — COMMENTED** [review]
Who can add product to whishlist. He must be buyer. How are you checking it. You are finding id only.

image

can you have consistent naming wishlistController instead of whishlist

When a product already exist in wishlist, you must display message to consider increading quantity

Resolve conflicts

**Comment by igorihimbazwe** [comment]
> Who can add product to whishlist. He must be buyer. How are you checking it. You are finding id only.
> 
> image
> 
> can you have consistent naming wishlistController instead of whishlist
> 
> When a product already exist in wishlist, you must display message to consider increading quantity
> 
> 
> 

thanks for you feedback @Celestin4 

ok on the role of the user i also proposed that but for now my user table does not have the role attribute that's why my PR is labeled as still in progress 


but on the quantity thing you've proposed, according to the research I've made there no quantity needed on wishlist, wishlist is like liking the product

**Review by UmuhireJessie — APPROVED** [review]
Great!

**Review by lilika67 — APPROVED** [review]

**Review by pacifiquemboni — APPROVED** [review]

**Review by jniyonkuru — APPROVED** [review]

File: src/server.ts [hunk 1]
```diff
@@ -38,8 +40,7 @@ app.use(passport.initialize());
 //Use body-parser middleware
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(bodyParser.json());
-app.use('/api', routes);
-app.use('/api', changePasswordIgnored, routes);
+app.use('/api', routes, authenticate, wishlistRoutes, changePasswordIgnored);
```
Comment (Celestin4) [inline src/server.ts]: This is not global middleware, it has to be imported in with wishlist routes

Comment (igorihimbazwe) [reply src/server.ts]: no the reason i made this global is because i wanted it to be used by anyone who want to check if a user is logged in

**Review by Celestin4 — CHANGES_REQUESTED** [review]
Check my comment

File: src/server.ts (hunk 1 above)
Comment (Celestin4) [reply src/server.ts]: why are not importing your routes in index.ts as we have ti import only one routes in server.ts

File: src/routes/wishlistRoute.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import express from 'express';
+import WishlistController from '../controllers/wishlist';
+
+const router = express.Router();
+
+router.post('/buyer/wishlist/:productId', WishlistController.addToWishlist);
```
Comment (Celestin4) [inline src/routes/wishlistRoute.ts]: How are you having multiple routes starting with /buyer

you have to implement relative routes

**Review by Celestin4 — APPROVED** [review]

**Pull request merged**